mimc_stark/test_recovery.py

```diff
@@ -66,7 +66,120 @@ def test_recovery():
         print("Recovery in dataset of size %i done in time %.4f" %
               (2**L, time.time() - a))
     print("Passed expansive test")
-    
+
+def is_power_of_two(x):
+    return x > 0 and x & (x-1) == 0
+
+def reverse_bit_order(n, order):
+    """
+    Reverse the bit order of an integer n
+    """
+    assert is_power_of_two(order)
+    # Convert n to binary with the same number of bits as "order" - 1, then reverse its bit order
+    return int(('{:0' + str(order.bit_length() - 1) + 'b}').format(n)[::-1], 2)
+
+
+def list_to_reverse_bit_order(l):
+    """
+    Convert a list between normal and reverse bit order. This operation is idempotent.
+    """
+    return [l[reverse_bit_order(i, len(l))] for i in range(len(l))]
+
+def test_recovery_danksharding():
+    modulus = 52435875175126190479447740508185965837690552500527637822603658699938581184513
+    nonresidue = 5
```

Comment:
What is this? If you hope this is a primitive root of unity, it is not (7 is)

Reply:
Yes, it should be a primitive root of unity, and I borrowed the number from kzg_data_availability here (https://github.com/ethereum/research/blob/a26cd10596a845040322576acbeb621c5040b483/kzg_data_availability/kzg_proofs.py#L6).  Could you please confirm which number should use (5 or 7 or either is OK?)

Reply:
No it is not (this one is a mistake, we should fix it).

Reply:
Thanks for pointing it out.  I have changed the value to 7.  Please let me know if you have any other questions.